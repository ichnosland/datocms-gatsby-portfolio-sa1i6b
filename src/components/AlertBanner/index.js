/**
*
* AlertBanner
*
*/

import styled, { css } from 'styled-components';
import { lighten, darken } from 'polished';
import FlexBox from '../FlexBox';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BannerStyle = css`
  ${({ tag }) => tag && css`display: inline-flex;`};
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  text-align: center;
  margin: ${(props) => props.margin ? props.margin : '0'};
  border: ${(props) => props.border ? `1px solid ${props.theme.brand}` : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  ${({ actioncolor }) => actioncolor && css`
    border: 1px solid ${colore.actions[actioncolor]};
    background-color: ${lighten(0.4, colore.actions[actioncolor])};
    color: ${darken(0.2, colore.actions[actioncolor])};
    svg {
      fill: ${darken(0.2, colore.actions[actioncolor])};
    }
  `}
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const AlertBanner = styled(FlexBox)`
  ${BannerStyle}
  padding: ${(props) => props.padding ? props.padding : '0.6em  1.6em'};
  svg {
    fill: ${({ color, actioncolor }) => {
    if (color) return color;
    if (actioncolor) return actioncolor;

    return colore.ui.txt;
  }};
  }
`;

AlertBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const InfoBanner = styled(AlertBanner)`
  padding: ${(props) => props.padding ? props.padding : '0.6em  1em'};
  ${media.lt667`
    flex-direction: column;
    svg {
      margin: 10px 0;
    }
  `}
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '0.6em'};
  `}
`;

InfoBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const IconBanner = styled(FlexBox)`
  ${BannerStyle}
  align-self: stretch;
  align-items: center;
  padding: ${(props) => props.padding ? props.padding : '9px 16px'};
  div {
    margin-right: 12px;
    align-items: center;
    &:last-child {
      margin-right: 0;
    }
  }
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '7px 12px'};
  `}
`;

IconBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export default AlertBanner;
