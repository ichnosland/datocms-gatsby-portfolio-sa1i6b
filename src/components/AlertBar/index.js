/**
*
* AlertBar
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import styled, { css } from 'styled-components';

import Div from '../Div';
import FlexBox from '../FlexBox';
import Svg from '../Svg';
import { IconBtn } from '../TopBar';
import { H3 } from '../Heading';
import media from 'style/mediainjector';
import { colore } from 'style/color';
import icon from 'icons/globals';
import { ContainerStyle } from '../Container';

export const AlertWrap = styled(Div)`
  position: absolute;
  top: 0;
  left: 0;
  color: ${colore.ui.contrast};
  width: 100%;
  height: 56px;
  padding: 0 20px;
  background-color: ${(props) => props.theme.brand};
  ${({ giusto }) => giusto && css`background-color: ${colore.actions.okay};`}
  ${({ sbagliato }) => sbagliato && css`background-color: ${colore.actions.error};`}
  ${({ aiuto }) => aiuto && css`background-color: ${colore.actions.help};`}
  z-index: 8;
  transition: all 0.5s;
  ${media.desktop`
    border-top-left-radius: ${(props) => props.versione ? '3px' : '20px'};
    border-top-right-radius: ${(props) => props.versione ? '3px' : '20px'};
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

AlertWrap.defaulProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const BarContainer = styled(FlexBox)`
  ${ContainerStyle}
  height: 100%;
  ${(props) => props.helpFx && css`
    justify-content: space-between;
    &::after {
      content: "";
      display: inline;
    }
  `}
  h3 {
    color: ${colore.ui.contrast};
  }
`;

class AlertBar extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      giusto,
      sbagliato,
      aiuto,
      versione,
      show,
      helpFx,
    } = this.props;
    return (
      <AlertWrap
        giusto={giusto}
        sbagliato={sbagliato}
        aiuto={aiuto}
        show={show}
        versione={versione}
      >
        <BarContainer helpFx={helpFx}>
          {helpFx &&
            <IconBtn onClick={helpFx}>
              <Svg {...icon.topBarFeedback} />
            </IconBtn>
          }
          <H3>
            {giusto &&
              'Risposta corretta!'
            }
            {sbagliato &&
              'Hai sbagliato!'
            }
            {aiuto &&
              'Aiuto'
            }
          </H3>
        </BarContainer>
      </AlertWrap>
    );
  }
}

AlertBar.propTypes = {
  giusto: PropTypes.bool,
  sbagliato: PropTypes.bool,
  aiuto: PropTypes.bool,
  versione: PropTypes.bool,
  show: PropTypes.bool.isRequired,
  helpFx: PropTypes.func,
};

export default AlertBar;
