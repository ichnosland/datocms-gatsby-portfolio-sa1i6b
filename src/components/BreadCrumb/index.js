/**
*
* BreadCrumb
*
*/

import styled, { css } from 'styled-components';

import Div from 'components/Div';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BreadCrumb = styled(Div)`
  color: ${(props) => props.color};
  position: relative;
  display: ${(props) => props.display ? props.display : 'inline-block'};
  text-align: center;
  border: ${(props) => props.border ? props.border : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const Crumb = styled(Div)`
  position: relative;
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  align-content: center;
  flex-wrap: nowrap;
  padding: 0.4em 1.4em 0.4em 0.8em;
  &:first-child {
    padding-left: 1.2em;
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:not(:first-child):not(:last-child) {
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:only-child {
    &:after {
      content: "";
      position: relative;
    }
  }
`;

BreadCrumb.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export default BreadCrumb;
