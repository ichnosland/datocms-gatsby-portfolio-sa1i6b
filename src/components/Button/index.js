/**
*
* Button
*
*/

import styled, { css } from 'styled-components';
//import { lighten, darken, transparentize } from 'polished';
import media from 'style/mediainjector';
import { colore, transparent } from 'style/color';
import Svg from '../Svg';

export const ButtonStyle = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  font-family: ${(props) => props.theme.fonts.buttons.fontFamily};
  font-weight: ${(props) => props.theme.fonts.buttons.fontWeight};
  ${({ full }) => full && 'width: 100% !important;'};
  ${({ half }) => half && 'width: 50% !important;'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  text-align: center;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding ? props.padding : '0.4em  1.4em'};
  border: ${(props) => props.border ? props.border : '1px solid transparent'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : props.theme.brand};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.buttons};
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : props.theme.brand))};
    color: ${(props) => transparentize(0.5, (props.color ? props.color : colore.ui.contrast))};
    pointer-events: none;
    svg {
        opacity: .3;
      }
  }
  ${({ round }) => round && css`
    padding: 0;
    width: ${(props) => props.width};
    height: ${(props) => props.width};
    border-radius: 100%;
  `}
  ${({ actioncolor }) => actioncolor && css`
    background-color: ${colore.actions[actioncolor]};
    &:disabled {
      background-color: ${lighten(0.3, colore.actions[actioncolor])};
      color: ${lighten(0.1, colore.actions[actioncolor])};
    }
  `}
  ${({ admin }) => admin && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${colore.ui.neutralBg};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
  `}
  ${({ outline }) => outline && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${(props) => props.theme.brand};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
    svg {
      fill: ${(props) => props.fill ? props.fill : props.theme.brand};
    }
  `}
  ${({ bone }) => bone && css`
    color: ${colore.ui.contrast};
    border: 2px solid ${colore.ui.contrast};
    background-color: ${transparent};
    &:hover,
    &:active {
      background-color: rgba(0,0,0,.1);
    }
  `}
  &:active,
  &:focus {
    outline: none;
  }
  ${(props) => props.standard && css`
    min-width: 180px;
    max-width: 320px;
  `};
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${media.iphone45`
    ${({ standard }) => standard && 'min-width: 140px;'}
    padding: ${(props) => props.padding ? props.padding : '0.4em  0.6em;'};
    svg {
      margin: 0 6px 0 0;
    }
  `}
`;

export const Button = styled.button`
  ${ButtonStyle}
`;

Button.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const SubmitButton = styled.input`
  ${ButtonStyle}
`;

SubmitButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const ActionButtonStyle = css`
  font-size: 21px;
  padding: 21px 30px;
  background-color: ${(props) => props.bgcolor};
  border: ${(props) => props.border ? props.border : '2px solid white'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  font-weight: 600;
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${({ sectionbutton }) => sectionbutton && css`
    min-width: 50%;
    font-size: 18px;
    padding: 14px 30px;
    ${media.lt667`
      width: 100%;
    `}
  ${media.lt480`
    font-size: 16px;
    `}
  `}
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : colore.landing.green))};
    color: ${(props) => transparentize(0.5, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
  }
`;

export const ActionButton = styled(Button)`
  ${ActionButtonStyle};
`;

export const Icon = styled(Svg)`
  width: ${(props) => props.size ? props.size : '18px'};
  height: ${(props) => props.size ? props.size : '18px'};
  margin: ${(props) => props.margin};
  fill: ${(props) => props.fill ? props.fill : colore.ui.contrast};
  ${({ right }) => right && css`
    margin: 0 -6px 0 6px;
  `}
  ${({ left }) => left && css`
    margin: 0 6px 0 -6px;
  `}
  ${({ brandColor }) => brandColor && css`
    fill: ${(props) => props.theme.brand};
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

Icon.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const NoStyleButton = css`
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const GhostButton = styled.button`
  ${NoStyleButton}
  display: inline-flex;
  align-items: center;
  justify-content: center;
  svg {
    fill: ${(props) => props.theme.brand};
  }
  ${(props) => props.fill && css`
    svg {
      fill: ${props.fill};
    }
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

GhostButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const TextButton = styled.button`
  color: ${(props) => props.color ? props.color : props.theme.brand};
  appearance: none;
  border: none;
  background: none;
  &:active,
  &:focus {
    outline: none;
  }
  &:hover {
    ${media.desktop`
      color: ${(props) => props.theme.darken};
    `}
  }
`;

TextButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    darken: '#1588AA',
  },
};
