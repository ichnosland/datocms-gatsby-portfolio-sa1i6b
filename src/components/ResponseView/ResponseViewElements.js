/**
*
* ResponseViewElements
*
*/

import styled, { css } from 'styled-components';

import FlexBox, { FlexChild } from '../FlexBox';
import Div from '../Div';
import LogoPlatform from '../LogoPlatform';
import { ContainerStyle } from '../Container';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const FeedbackSteps = styled(Div)`
  display: inline-block;
  text-align: center;
  margin-bottom: 20px;
  p {
    margin-bottom: 0;
  }
  ul {
    list-style: none;
    display: inline-block;
    padding: 0;
    text-align: center;
  }
  li {
    color: ${colore.ui.contrast};
    display: inline-block;
    width: 34px;
    height: 34px;
    margin: 2px;
    padding: 8px;
    border-radius: 3px;
    transition: all 1s linear 0s;
    &.stepsDone {
      color: ${colore.actions.okay};
      background-color: ${colore.ui.mainBg};
    }
    ${(props) => props.stepsDone && css`
      color: ${colore.actions.okay};
      background-color: ${colore.ui.mainBg};
    `}
  }
  ${media.mobile`
    width: 100%;
    margin-bottom: 0;
  `}
`;

export const ResponseWrap = styled(FlexBox)`
  color: ${colore.ui.contrast};
  height: 100%;
  flex-direction: column;
  background: ${({ votoFinale }) => votoFinale >= 6 ?
    colore.actions.okay
    :
    '#ed5b53'};
`;

export const ResponseContainer = styled(FlexChild)`
  ${ContainerStyle}
  img {
    width: 50%;
  }
  ${media.lt480`
    flex-direction: column;
  `}
`;

export const ResponseBanner = styled(Div)`
  width: 50%;
  text-align: center;
  ${media.lt480`
    width: 100%;
  `}
  h2 {
    font-size: 3em;
    margin: 0.3em auto;
  }
`;

export const ResponseLogo = styled(LogoPlatform)`
  ${media.lt480`
    display: none;
  `}
`;
