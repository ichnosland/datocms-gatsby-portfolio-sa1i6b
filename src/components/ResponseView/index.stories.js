import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select, number, boolean } from '@storybook/addon-knobs';
import { ThemeProvider } from 'styled-components';
import 'global-styles';
import { allThemes, themeOptions } from 'style/theme';
import { ResponseView } from './index';
import goodBg from './images/good.gif';
import badBg from './images/bad.gif';

storiesOf('ResponseView', module)
  .addDecorator(withKnobs)
  .addWithJSX('ResponseView', () => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <ResponseView
        votoApprossimato={number('votoApprossimato', 7)}
        goodSrc={goodBg}
        badSrc={badBg}
        steps={boolean('steps', false)}
        lezioniCompletate={number('lezioniCompletate', 2)}
        lezioniTotali={number('lezioniTotali', 7)}
        resetFunction={() => { }}
        product="alatin"
      />
    </ThemeProvider>
  ), {
    skip: 1,
  });
