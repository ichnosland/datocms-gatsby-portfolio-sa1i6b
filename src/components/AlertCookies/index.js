/**
*
* AlertCookies
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import P from 'components/Paragraph';
import { Button } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const AlertCookiesWrap = styled(Div)`
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 0 20px 20px;
  z-index: 1;
  ${media.lt667`
    padding: 0 10px 30px;
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

export const CookiesBanner = styled(Div)`
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  max-width: 860px;
  margin: 0 auto;
  font-size: 14px;
  border: ${(props) => props.border};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : 'rgba(255,255,255,.95)'};
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,.15);
  box-shadow: 0 2px 1px rgba(0,0,0,.25);
  p {
    padding: 20px;
    margin: 0;
    &:last-child {
      padding: 0 20px 10px;
      text-align: right;
      ${media.lt480`
        padding: 0 20px 20px;
        text-align: center;
      `}
    }
  }
  ${media.lt667`
    font-size: 12px;
  `}
`;

class AlertCookies extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { show, color, border, bgcolor, onClickFunction } = this.props;
    return (
      <AlertCookiesWrap show={show}>
        <CookiesBanner
          color={color}
          border={border}
          bgcolor={bgcolor}
        >
          <P>Al fine di offrire il miglior servizio possibile questo sito utilizza cookies tecnici e di terze parti per funzionalità quali l&apos;accesso tramite social network e/o la visualizzazione di media. Se non acconsenti all&apos;utilizzo dei cookie di terze parti, alcune di queste funzionalità potrebbero essere non disponibili. Chiudendo questo banner o proseguendo con la navigazione acconsentirai al loro impiego. Per saperne di pi&ugrave; o negare il consenso a tutti o ad alcuni cookie puoi consultare <a href="https://maieuticallabs.it/privacy" target="_blank"><span>l&apos;informativa sulla privacy</span></a>.</P>
          <P><Button actioncolor="hint" standard={1} onClick={onClickFunction}>Chiudi</Button></P>
        </CookiesBanner>
      </AlertCookiesWrap>
    );
  }
}

AlertCookies.propTypes = {
  color: PropTypes.string,
  border: PropTypes.string,
  bgcolor: PropTypes.string,
  show: PropTypes.bool.isRequired,
  onClickFunction: PropTypes.func,
};

export default AlertCookies;
