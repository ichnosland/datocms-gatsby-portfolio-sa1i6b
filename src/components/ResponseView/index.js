/**
*
* ResponseView
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import { Button } from '../Button';
import { colore } from 'style/color';
import { H2 } from '../Heading';

import {
  FeedbackSteps,
  ResponseWrap,
  ResponseBanner,
  ResponseContainer,
  ResponseLogo,
} from './ResponseViewElements';

export class ResponseView extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      votoApprossimato,
      goodSrc,
      badSrc,
      resetFunction,
      product,
      nascondiVoto,
      titolo,
      steps,
      lezioniTotali,
      lezioniCompletate,
    } = this.props;

    let titoloVisualizzato = '';
    if (!nascondiVoto) {
      titoloVisualizzato = 'Il tuo voto Ã¨';
    } else if (!steps) {
      titoloVisualizzato = titolo;
    }

    return (
      <ResponseWrap
        votoFinale={votoApprossimato}
      >
        <ResponseContainer>
          {votoApprossimato >= 6 ?
            <img src={goodSrc} alt="Ottimo" /> :
            <img src={badSrc} alt="Risultato insufficiente" />
          }
          <ResponseBanner>
            {steps &&
              <FeedbackSteps>
                <p>Hai completato {lezioniCompletate} di {lezioniTotali} lezioni</p>
                <ul>
                  {Array.from({ length: lezioniTotali }, (_, i) => (
                    <li key={`li_${i}`} className={lezioniCompletate >= (i + 1) ? 'stepsDone' : ''}>{i + 1}</li>
                  ))}
                </ul>
              </FeedbackSteps>
            }
            <p>{titoloVisualizzato}</p>
            {!nascondiVoto && <H2 color={colore.ui.contrast}>{votoApprossimato}</H2>}
            <Button
              bone={1}
              standard
              onClick={resetFunction}
            >Ok
            </Button>
          </ResponseBanner>
        </ResponseContainer>
        <ResponseLogo width="100%" inverse product={product} execLogo />
      </ResponseWrap>
    );
  }
}

ResponseView.propTypes = {
  votoApprossimato: PropTypes.number.isRequired,
  goodSrc: PropTypes.string,
  badSrc: PropTypes.string,
  product: PropTypes.string,
  steps: PropTypes.bool,
  resetFunction: PropTypes.func.isRequired,
  nascondiVoto: PropTypes.bool,
  titolo: PropTypes.string,
};

export default ResponseView;
