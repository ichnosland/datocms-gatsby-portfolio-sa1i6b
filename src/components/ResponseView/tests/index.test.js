import React from 'react';
import { shallow } from 'enzyme';

import { Button } from 'components/Button';
import ResponseView from '../index';
import { ResponseWrap } from '../ResponseViewElements';
import goodBg from '../images/good.gif';
import badBg from '../images/bad.gif';

const mockProps = {
  votoApprossimato: 7,
  resetFunction: () => { },
  goodSrc: goodBg,
  badSrc: badBg,
  product: 'alatin',
};

describe('<ResponseView />', () => {
  it('should render a ResponseWrap', () => {
    const renderedComponent = shallow(
      <ResponseView {...mockProps} />
    );
    expect(renderedComponent.find(ResponseWrap).length).toBe(1);
    expect(renderedComponent.find(Button).length).toBe(1);
    expect(renderedComponent.find('img').length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('restituisce badSrc se votoApprossimato <= 6', () => {
    const myProps = {
      ...mockProps,
      votoApprossimato: 4,
    };
    const renderedComponent = shallow(
      <ResponseView {...myProps} />
    );
    expect(renderedComponent.find(Button).length).toBe(1);
    expect(renderedComponent.find('img').length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('testo il caso delle unitÃ ', () => {
    const myProps = {
      ...mockProps,
      votoApprossimato: 10,
      nascondiVoto: true,
      titolo: 'Titolo responso',
      lezioniCompletate: 2,
      lezioniTotali: 7,
      steps: true,
    };
    const renderedComponent = shallow(
      <ResponseView {...myProps} />
    );
    expect(renderedComponent.find(Button).length).toBe(1);
    expect(renderedComponent.find('img').length).toBe(1);
    expect(renderedComponent.find('ul').length).toBe(1);
    expect(renderedComponent.find('li').length).toBe(7);
    expect(renderedComponent.find('li.stepsDone').length).toBe(2);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('testo il caso in cui !steps e titolo settato', () => {
    const myProps = {
      ...mockProps,
      votoApprossimato: 10,
      nascondiVoto: true,
      titolo: 'Titolo responso',
      lezioniCompletate: 2,
      lezioniTotali: 7,
      steps: false,
    };
    const renderedComponent = shallow(
      <ResponseView {...myProps} />
    );
    expect(renderedComponent.find(Button).length).toBe(1);
    expect(renderedComponent.find('img').length).toBe(1);
    expect(renderedComponent.find('ul').length).toBe(0);
    expect(renderedComponent.find('li').length).toBe(0);
    expect(renderedComponent.find('li.stepsDone').length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });
});
