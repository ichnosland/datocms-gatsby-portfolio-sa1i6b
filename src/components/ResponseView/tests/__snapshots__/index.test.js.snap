// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponseView /> restituisce badSrc se votoApprossimato <= 6 1`] = `
<ResponseViewElements__ResponseWrap
  votoFinale={4}
>
  <ResponseViewElements__ResponseContainer>
    <img
      alt="Risultato insufficiente"
      src="IMAGE_MOCK"
    />
    <ResponseViewElements__ResponseBanner>
      <p>
        Il tuo voto è
      </p>
      <Heading__H2
        color="rgb(255, 255, 255)"
      >
        4
      </Heading__H2>
      <Button
        actioncolor=""
        bone={1}
        onClick={[Function]}
        standard={true}
        theme={
          Object {
            "brand": "#00BBEF",
            "fonts": Object {
              "buttons": Object {
                "fontFamily": "inherit",
                "fontWeight": "inherit",
              },
            },
            "radius": Object {
              "buttons": "6px",
            },
          }
        }
      >
        Ok
      </Button>
    </ResponseViewElements__ResponseBanner>
  </ResponseViewElements__ResponseContainer>
  <ResponseViewElements__ResponseLogo
    execLogo={true}
    inverse={true}
    product="alatin"
    width="100%"
  />
</ResponseViewElements__ResponseWrap>
`;

exports[`<ResponseView /> should render a ResponseWrap 1`] = `
<ResponseViewElements__ResponseWrap
  votoFinale={7}
>
  <ResponseViewElements__ResponseContainer>
    <img
      alt="Ottimo"
      src="IMAGE_MOCK"
    />
    <ResponseViewElements__ResponseBanner>
      <p>
        Il tuo voto è
      </p>
      <Heading__H2
        color="rgb(255, 255, 255)"
      >
        7
      </Heading__H2>
      <Button
        actioncolor=""
        bone={1}
        onClick={[Function]}
        standard={true}
        theme={
          Object {
            "brand": "#00BBEF",
            "fonts": Object {
              "buttons": Object {
                "fontFamily": "inherit",
                "fontWeight": "inherit",
              },
            },
            "radius": Object {
              "buttons": "6px",
            },
          }
        }
      >
        Ok
      </Button>
    </ResponseViewElements__ResponseBanner>
  </ResponseViewElements__ResponseContainer>
  <ResponseViewElements__ResponseLogo
    execLogo={true}
    inverse={true}
    product="alatin"
    width="100%"
  />
</ResponseViewElements__ResponseWrap>
`;

exports[`<ResponseView /> testo il caso delle unità 1`] = `
<ResponseViewElements__ResponseWrap
  votoFinale={10}
>
  <ResponseViewElements__ResponseContainer>
    <img
      alt="Ottimo"
      src="IMAGE_MOCK"
    />
    <ResponseViewElements__ResponseBanner>
      <ResponseViewElements__FeedbackSteps>
        <p>
          Hai completato 
          2
           di 
          7
           lezioni
        </p>
        <ul>
          <li
            className="stepsDone"
            key="li_0"
          >
            1
          </li>
          <li
            className="stepsDone"
            key="li_1"
          >
            2
          </li>
          <li
            className=""
            key="li_2"
          >
            3
          </li>
          <li
            className=""
            key="li_3"
          >
            4
          </li>
          <li
            className=""
            key="li_4"
          >
            5
          </li>
          <li
            className=""
            key="li_5"
          >
            6
          </li>
          <li
            className=""
            key="li_6"
          >
            7
          </li>
        </ul>
      </ResponseViewElements__FeedbackSteps>
      <p />
      <Button
        actioncolor=""
        bone={1}
        onClick={[Function]}
        standard={true}
        theme={
          Object {
            "brand": "#00BBEF",
            "fonts": Object {
              "buttons": Object {
                "fontFamily": "inherit",
                "fontWeight": "inherit",
              },
            },
            "radius": Object {
              "buttons": "6px",
            },
          }
        }
      >
        Ok
      </Button>
    </ResponseViewElements__ResponseBanner>
  </ResponseViewElements__ResponseContainer>
  <ResponseViewElements__ResponseLogo
    execLogo={true}
    inverse={true}
    product="alatin"
    width="100%"
  />
</ResponseViewElements__ResponseWrap>
`;

exports[`<ResponseView /> testo il caso in cui !steps e titolo settato 1`] = `
<ResponseViewElements__ResponseWrap
  votoFinale={10}
>
  <ResponseViewElements__ResponseContainer>
    <img
      alt="Ottimo"
      src="IMAGE_MOCK"
    />
    <ResponseViewElements__ResponseBanner>
      <p>
        Titolo responso
      </p>
      <Button
        actioncolor=""
        bone={1}
        onClick={[Function]}
        standard={true}
        theme={
          Object {
            "brand": "#00BBEF",
            "fonts": Object {
              "buttons": Object {
                "fontFamily": "inherit",
                "fontWeight": "inherit",
              },
            },
            "radius": Object {
              "buttons": "6px",
            },
          }
        }
      >
        Ok
      </Button>
    </ResponseViewElements__ResponseBanner>
  </ResponseViewElements__ResponseContainer>
  <ResponseViewElements__ResponseLogo
    execLogo={true}
    inverse={true}
    product="alatin"
    width="100%"
  />
</ResponseViewElements__ResponseWrap>
`;
